xwiki-platform-core/xwiki-platform-rendering/xwiki-platform-rendering-macros/xwiki-platform-rendering-macro-include/src/main/java/org/xwiki/rendering/internal/macro/include/IncludeMacro.java

```diff
@@ -202,7 +203,20 @@ public void setDocumentDisplayer(DocumentDisplayer documentDisplayer)
                 references.pop();
             }
         }
-
+        
+        // Exclude the first heading from the content
+        if (parameters.excludeFirstHeading()) {
+            HeaderBlock heading = result.getFirstBlock(new ClassBlockMatcher(HeaderBlock.class),
+                    Block.Axes.DESCENDANT_OR_SELF);
```

Comment:
BTW this is wrong. You need to write some test cases to figure it out. Try different examples you can think of content with headings and I'm sure you'll see why.

Reply:
I tested in all 5 cases but I didn't find any bug in my implemented examples. I can also use `getBlocks()` function to get the heading. But this single line approach was good to use.